src/runtime/string.go

```diff
@@ -283,3 +283,15 @@ func cgo_GoBytes(ptr unsafe.Pointer, length uintptr) []byte {
 	}
 	return buf
 }
+
+func cgo_CBytes(b []byte) unsafe.Pointer {
+	p := malloc(uintptr(len(b)))
+	sliceHeader := struct {
+		p   unsafe.Pointer
+		len uintptr
+		cap uintptr
+	}{p, uintptr(len(b)), uintptr(len(b))}
+	s := *(*[]byte)(unsafe.Pointer(&sliceHeader))
```

Comment:
Sorry I realized this only now but there is a much cleaner way to do this:

```suggestion
    s := unsafe.Slice(p, len(b))
```